tensorboard/util/test_util.py

```diff
@@ -273,3 +278,93 @@ def ensure_tb_summary_proto(summary):
     return summary
 
   return summary_pb2.Summary.FromString(summary.SerializeToString())
+
+
+def run_v1_only(reason, func=None):
+  """Execute the decorated test only if running in v1 mode.
+
+  This function is intended to be applied to tests that exercise v1 only
+  functionality. If the test is run in v2 mode it will simply be skipped.
+
+  Args:
+    reason: string giving a reason for limiting the test to v1 only. No
+      component should be TensorFlow only, moreover v1 only.
+    func: function to be annotated. If `func` is None, this method returns a
+      decorator the can be applied to a function. If `func` is not None this
+      returns the decorator applied to `func`.
+
+  Returns:
+    Returns a decorator that will conditionally skip the decorated test method.
+  """
+
+  def decorator(f):
+    if inspect.isclass(f):
+      setup = f.__dict__.get("setUp")
```

Comment:
I’m not sure exactly what to make of this `setUp` patching. On the one
hand, `setUp` only runs if an actual test case runs, and we patch all
the test cases, so this seems a bit superfluous. On the other hand, due
to test case inheritance, (a) you could have more test cases on a
subclass, and (b) `__dict__` is not sufficient to patch `setUp` because
the `setUp` in question could be defined on a superclass (possibly using
subclass-dependent method implementations). I _think_ we’re safe here
because logically there should be an independence principle (if your
test class extends a test class that’s V1-only, then you must also be
V1-only). Is this consistent with your reasoning? If not, can you
explain why you believe that this mechanism for `setUp` patching is
correct?

Reply:
This was borrowed code in verbatim from TF so I did not have any rationale in doing so.
>setUp only runs if an actual test case runs

If this is true, I do agree that this patching is quite superfluous  and I wonder if TF ever had a case when a class is v1 only and one method is v2 only. This seems very unpredictable (how does test author know if setUp gets run? and also, why is setUp skipped but not tearDown?) and should be warned against (since v2 only will be ineffective AFAICT).

Anyways, I will remove this.

Reply:
> This seems very unpredictable (how does test author know if setUp gets
> run? and also, why is setUp skipped but not tearDown?)

I must be missing something. It seems very predictable to me. The logic is:

 1. `setUpClass` runs
 2. for each test case:
     1. `setUp` runs (skip next two steps if this fails)
     2. the test runs
     3. `tearDown` runs
 3. `tearDownClass` runs

It follows naturally that `setUp` only runs if a test case runs, just
from the semantics of what `setUp` means (“runs before each test”).